### PR TITLE
Issue #587 Change type of category in Mail.add_category from string to Category

### PR DESCRIPTION
**Pull request by lifez**
# Fixes # 
#587

### Checklist
- [x] I have made a material change to the repo (functionality, testing, spelling, grammar)
- [x] I have read the [Contribution Guide] and my PR follows them.
- [x] I updated my branch with the master branch.
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation about the functionality in the appropriate .md file
- [x] I have added in line documentation to the code I modified

### Short description of what this PR does:
- Change type of category in Mail.add_category from string to Category to prevent warning from some editors

### CODE REVIEW

**Comment by SendGridDX** [comment]
[CLA assistant check](https://cla.sendgrid.com/sendgrid/sendgrid-python?pullRequest=595) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.sendgrid.com/sendgrid/sendgrid-python?pullRequest=595) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.sendgrid.com/check/sendgrid/sendgrid-python?pullRequest=595) it.</sub>

**Comment by lifez** [comment]
How I can rebuild again without any commit?

**Comment by thinkingserious** [comment]
No need to worry about that, we'll take care of it in the code review. Thanks!

File: sendgrid/helpers/mail/mail.py [hunk 1]
```diff
@@ -376,9 +376,9 @@ def categories(self):
         return self._categories
 
     def add_category(self, category):
-        """Add a Category to this Mail.  Must be less than 255 characters.
```
Comment (af4ro) [inline sendgrid/helpers/mail/mail.py]: What was your reason behind this deletion?

File: sendgrid/helpers/mail/mail.py [hunk 2]
```diff
@@ -376,9 +376,9 @@ def categories(self):
...
+        """Add a Category to this Mail.
 
-        :type category: string
+        :type category: Category
```
Comment (af4ro) [inline sendgrid/helpers/mail/mail.py]: This is correct, thank you for pointing it out!

File: sendgrid/helpers/mail/mail.py (hunk 1 above)
Comment (lifez) [reply sendgrid/helpers/mail/mail.py]: Since Category is a Category type `Must be less than 255 characters.` is not suitable.

**Comment by thinkingserious** [comment]
Hello @lifez,<br /><br/>Thanks again for the PR!<br />                                              <br />We want to show our appreciation by sending                                              you some swag. Could you please fill out                                              <a href="https://dx.sendgrid.com/swag">this form</a> so we can send it to you? Thanks!                                             <br /><br />Team SendGrid DX

**Pull request merged**